### DISCUSSION TITLE
Need to write files inside airflow but have Read-only error

### DISCUSSION THREAD

**Discussion by cynde** [apache/airflow#28214] [Q&A]
I need to write output files from some process in my airflow but it returns WARNING - OSError: [Errno 30] Read-only file system: '/usr/local/airflow/dags/datasets/1'. Already tried chmod

can somebody help me?

**Comment by Taragolis** [comment]
A bit small information about deployment of airflow? But due to the message you have Read-Only filesystem, so `chmod` wouldn't help.

And usual it is not a good idea change anything in DAG directory inside airflow tasks:
- You could overwrite some DAG or important file
- In general if it not shared mount (e.g. NFS share) this changes not appear in other workers
- Some dag synchronisation process might overwrite this changes

**Reply by cynde** [reply]
thank you for your response. so if I want to write a file or unzip a zip file then save it to a path, what should i do?

**Reply by potiuk** [reply]
Generally the same what you do for all other apps. There is nothing specific to airflow - I think if you are unable to determine it on your own and expect instructions, you have a huge knowledge gap that will block you from doing anything. Probably Airlfow forum is not the best place to learn the basics like that.

There many guidelines and online sources for making sure you can run applications on Linux and you should start from that.
